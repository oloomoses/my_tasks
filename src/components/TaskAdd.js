import React from 'react';

const TaskAdd = () => (
  <form>
    <input
      type="text"
      className="task-input"
      name="text"
      autoComplete="off"
      placeholder="Task Name"
    />

    <input
      type="text"
      className="task-input"
      name="text"
      placeholder="Task Description"
      autoComplete="off"
    />

    <button type="submit" className="btn">
      ADD
    </button>
  </form>
);

export default TaskAdd;
