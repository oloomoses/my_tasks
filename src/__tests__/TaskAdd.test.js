import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TaskAdd from '../components/TaskAdd';

let wrapper;
let addTask;
beforeEach(() => {
  addTask = (TaskName, TaskDescription, Avatar) => ({ TaskName, TaskDescription, Avatar });
  wrapper = shallow(<BrowserRouter><TaskAdd addTask={addTask} /></BrowserRouter>);
});

test('has a form', () => {
  expect(wrapper.find('form')).toBeTruthy();
});

test('has a submit button', () => {
  expect(wrapper.find('button')).toBeTruthy();
});

it('matches the snapshot', () => {
  const tree = renderer.create(
    <BrowserRouter><TaskAdd addTask={addTask} /></BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
