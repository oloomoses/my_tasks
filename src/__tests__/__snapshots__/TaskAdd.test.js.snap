// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoComplete="off"
    className="task-input"
    name="text"
    onChange={[Function]}
    placeholder="Task Name"
    required={true}
    type="text"
    value=""
  />
  <input
    autoComplete="off"
    className="task-input"
    name="text"
    onChange={[Function]}
    placeholder="Task Description"
    required={true}
    type="text"
    value=""
  />
  <input
    autoComplete="off"
    className="task-input"
    name="text"
    onChange={[Function]}
    placeholder="Avatar url"
    type="text"
    value=""
  />
  <button
    className="btn"
    type="submit"
  >
    ADD
  </button>
</form>
`;
